cpp/dolfin/io/HDF5File.cpp

```diff
@@ -469,25 +469,11 @@ void HDF5File::write(const mesh::MeshFunction<bool>& meshfunction,
   // so copy to int with values 1 and 0
   mesh::MeshFunction<int> mf(mesh, cell_dim, -1);
   for (auto& cell : mesh::MeshRange<mesh::MeshEntity>(*mesh, cell_dim))
-    mf[cell.index()] = (meshfunction[cell.index()] ? 1 : 0);
+    mf.values()[cell.index()] = (meshfunction.values()[cell.index()] ? 1 : 0);
```

Comment:
Avoid calling `mf.values()` inside loops (quite a few occurrences of this in the PR).

In this particular case, `mf.values() = meshfunction.values()` might even work, or some other Eigen syntax that doesn't need a loop.

Reply:
I moved `.values()` call outside of hot loop now.